### PR TITLE
CLOUDP-262219: AKO Backup Compliance Policy Export

### PR DESCRIPTION
**Pull request by roothorp**
## Proposed changes

Update the AKO project exporter to include backup compliance policies.

Note that this PR is branched from the [currently open 2.4 PR](https://github.com/mongodb/mongodb-atlas-cli/pull/3116), as it requires some of the changes & updates included there. I think GitHub reacts appropriately should that PR be merged, but I will keep an eye on it.

_Jira ticket:_ CLOUDP-211174

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by s-urbaniak — DISMISSED** [review]
lgtm, very cool 👏

File: internal/kubernetes/operator/project/project.go [hunk 1]
```diff
@@ -198,6 +200,18 @@ func BuildAtlasProject(br *AtlasProjectBuildRequest) (*AtlasProjectResult, error
...
+		bcp, err := buildBCP(br.ProjectStore, br.Project.Name, br.ProjectID, br.TargetNamespace, br.Version, br.Dictionary)
+		if err != nil {
+			return nil, err
+		}
+		bcpRef.Name = bcp.Name
+		bcpRef.Namespace = bcp.Namespace
+		result.BCP = bcp
+		projectResult.Spec.BackupCompliancePolicyRef = &bcpRef
+	}
+
```
Comment (josvazg) [inline internal/kubernetes/operator/project/project.go:214]: cc @filipcirtog we might need something like this for #3100 ?

**Review by josvazg — COMMENTED** [review]
LGTM, have we also done some manual testing on this using the compiled CLI binary by hand?

**Comment by helderjs** [comment]
I have started kube test to see everything is good

**Review by carriecwk — COMMENTED** [review]
LGTM

**Review by carriecwk — DISMISSED** [review]
LGTM

File: internal/kubernetes/operator/project/project.go (hunk 1 above)
Comment (filipcirtog) [reply internal/kubernetes/operator/project/project.go:214]: Yes! Great idea! Just seen that this Feature validator is also used for the data federation export!Thank you!

File: internal/kubernetes/operator/project/project.go [hunk 2]
```diff
@@ -198,6 +200,18 @@ func BuildAtlasProject(br *AtlasProjectBuildRequest) (*AtlasProjectResult, error
 		result.Teams = teams
 	}
 
+	if br.Validator.FeatureExist(features.ResourceAtlasProject, featureBCP) {
```
Comment (filipcirtog) [inline internal/kubernetes/operator/project/project.go:203]: Nice you used the validator! I'll also add this in my PR. Thank you for the idea!

File: internal/kubernetes/operator/project/project.go [hunk 3]
```diff
@@ -971,6 +985,54 @@ func buildTeams(teamsProvider store.OperatorTeamsStore, orgID, projectID, projec
 	return teamsRefs, atlasTeamCRs, nil
 }
 
+func buildBCP(bcpProvider store.CompliancePolicyDescriber, projectName, projectID, targetNamespace, version string, dictionary map[string]string) (*akov2.AtlasBackupCompliancePolicy, error) {
+	bcp, err := bcpProvider.DescribeCompliancePolicy(projectID)
+	if err != nil {
+		return nil, err
```
Comment (filipcirtog) [inline internal/kubernetes/operator/project/project.go:993]: Very straightforward approach! Got the object from the describer and populated the yaml with its fields.

**Review by filipcirtog — APPROVED** [review]
LGTM! Great job and thanks for the idea with the featureValidator!

**Review by josvazg — APPROVED** [review]

**Review by jeroenvervaeke — APPROVED** [review]

**Pull request merged**